fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBDatabase.java

```diff
@@ -1193,31 +1193,44 @@ public <T> T get(CompletableFuture<T> future) throws InterruptedException, Execu
     @VisibleForTesting
     @SuppressWarnings({"PMD.CloseResource", "PMD.GuardLogStatement"})
     public int warnAndCloseOldTrackedOpenContexts(long minAgeSeconds) {
-        long nanoTime = System.nanoTime() - TimeUnit.SECONDS.toNanos(minAgeSeconds);
-        if (trackedOpenContexts.isEmpty()) {
-            return 0;
-        }
-        try {
-            if (trackedOpenContexts.firstKey() > nanoTime) {
-                return 0;
-            }
-        } catch (NoSuchElementException ex) {
+        long cutoffTime = System.nanoTime() - TimeUnit.SECONDS.toNanos(minAgeSeconds);
+        @Nullable Map.Entry<Long, FDBRecordContext> firstEntry = trackedOpenContexts.firstEntry();
+        if (firstEntry == null || firstEntry.getKey() > cutoffTime) {
```

Comment:
This requires a bit of explanation: `SortedMap.firstKey` returns the lowest key or throws `NoSuchElementException` if it's empty. That's why the old code needed to catch `NoSuchElementException`. However, `NavigableMap.firstEntry` returns `null` if the map is empty, so the new code just does a null check, allowing us to avoid catching here. This is a somewhat muddled API, so I assume different methods were created at different times with somewhat confusing semantics